Services/ApartmentService.cs

```diff
@@ -0,0 +1,161 @@
+﻿using DTO.InputDTOs;
+using DTO.OutputDTOs;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Services
+{
+    public class ApartmentService : IApartmentService
+    {
+        private readonly IDbContextFactory<DatabaseContext> contextFactory;
+
+        public ApartmentService(IDbContextFactory<DatabaseContext> contextFactory)
+        {
+            this.contextFactory = contextFactory;
+        }
+
+        public async Task<Apartment?> GetAsync(int id)
+        {
+            using (DatabaseContext ctx = contextFactory.CreateDbContext())
+                return await ctx.Apartments
+                    .Include(x => x.BedPrizes)
+                    .Include(x => x.ApartmentFeatures)
+                    .FirstOrDefaultAsync(x => x.ID == id);
+        }
+
+        public async Task<IActionResult> CreateAsync(ApartmentDTO apartmentDto)
+        {
+            using(DatabaseContext ctx = contextFactory.CreateDbContext())
+            {
+                var landlord = await ctx.Landlords.FirstOrDefaultAsync(x => x.ID == apartmentDto.LandlordID);
+
+                if (landlord == null)
+                    return new UnauthorizedObjectResult(apartmentDto.LandlordID);
+
+                var newApartment = new Apartment()
```

Comment:
It would be a good idea to check if an apartment with such Name, MaxBedNumbers, City and Landlord does not exist